streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/trait/FlinkClientTrait.scala

```diff
@@ -115,6 +115,7 @@ trait FlinkClientTrait extends Logger {
       flinkConfig.setBoolean(
         SavepointConfigOptions.SAVEPOINT_IGNORE_UNCLAIMED_STATE,
         submitRequest.allowNonRestoredState)
+      flinkConfig.setString(RestoreMode.RESTORE_MODE, submitRequest.restoreMode.getName);
```

Comment:
Here, we should check the version number of Flink and then decide whether this parameter is needed or not.

Reply:
done